### ISSUE TITLE
RC 3 update breaks working Webpack production build

### ISSUE THREAD

**Issue by podlebar** [vuejs/vue-cli#1617]
### Version
3.0.0-rc.3

### Reproduction link
[https://gist.github.com/podlebar/4f8e44fd2eea8b586632507a7de0066e](https://gist.github.com/podlebar/4f8e44fd2eea8b586632507a7de0066e)

### Steps to reproduce
Update all deps to RC 3, run 'npm run build' or 'yarn build'.

packacke dev deps should be like:

```
"devDependencies": {
    "@vue/cli-plugin-eslint": "^3.0.0-rc.3",
    "@vue/cli-plugin-pwa": "^3.0.0-rc.3",
    "@vue/cli-service": "^3.0.0-rc.3",
    "@vue/eslint-config-prettier": "^3.0.0-rc.3",
    "node-sass": "^4.9.0",
    "sass-loader": "^7.0.1",
    "vue-template-compiler": "^2.5.16"
  },
```

### What is expected?
output should be somehow like:
```
  dist/js/chunk-vendors.81a28aea.js         274.86 kb        71.06 kb
  dist/js/app.314cd380.js                   8.86 kb          3.42 kb
  dist/precache-manifest.2419737b5bfdcdd    0.46 kb          0.22 kb
  5de20cb9b91a848a3.js
  dist/sw.js                                0.15 kb          0.14 kb
  dist/css/chunk-vendors.271358e2.css       14.13 kb         2.72 kb
  dist/css/app.2973a23a.css
```

also no service worker is created nor the manifest file.

### What is actually happening?
output is:
```
dist/app.js    1302.84 kb                    206.43 kb
```

---
It seems that the RS 3 update changes how Webpack build the production build.
I tried different settings and none of them worked. Also tried re-configure via VUE UI but no affect on production build.

Inspection Webpack output from VUE UI: https://gist.github.com/podlebar/4f8e44fd2eea8b586632507a7de0066e

**Comment by LinusBorg** [comment]
Please provide a repository with a runable reproduction.

**Comment by podlebar** [comment]
@LinusBorg 
https://github.com/podlebar/vue-cli-rc3-issue

updated from 3.0.0-beta.15

**Comment by LinusBorg** [comment]
Can't reproduce the problem, everthing's where it should be when I run `build`.

![image](https://user-images.githubusercontent.com/1444526/41645078-358662e0-7470-11e8-8234-5f06c5424729.png)

Have you tried deleting `/node_modules` and re-installing all dependencies?

**Comment by podlebar** [comment]
yes.. first thing i tried.. i realized now that if a create a new project via cli the same things happen.
Uninstalling vue-cli and reinstall did not help.. any other global module i should try to remove?

i tried 'npm uninstall -g @vue/cli' but maybe there is more to to for a 'reset'.

**Comment by LinusBorg** [comment]
> i tried 'npm uninstall -g ' but maybe there is more to to for a 'reset'.

No, `npm uninstall -g @vue/cli && npm i -g @vue-cli` should suffice

**Comment by podlebar** [comment]
very strange.. reinstalled everything and still i get the same..
also i noticed that on 'npm run build' i get 'Images and other types of assets omitted.'. So for some reason the CSS not touched at all but the webpack config clearly contains a rule for css: https://gist.github.com/podlebar/c10cd6e14545ad6f29780b258122a33c

**Comment by podlebar** [comment]
should there not be something like ExtractTextPlugin to get the css?

**Comment by LinusBorg** [comment]
Hm, it seems that it's running the `build` command with `NODE_ENV=development`  as environment variables.

that would explain why there's still vue-style-loader in there instad of the extract plugin etc.

Please run `printenv` in your terminal (ssuming your are on OSX/linux) and check if you have NODE_ENV set as a permanent environment variable...

**Comment by podlebar** [comment]
yes i have NODE_ENV=local .. but this is since quiet some time like this..
i removed it and reloaded my .zshrc .. no affect.. but it worked after restarting iterm2..

was that handled differently in the previous releases?

@LinusBorg thx a lot for the help.. you can kindly close the issue

**Comment by LinusBorg** [comment]
> was that handled differently in the previous releases?

In a way, yes. in beta.16 we made sure that variables in .env vars don't overwrite variables set externally - this change allows you to overwrite environment variables by defining them in your CI environment, without the need for a separate .env file for it.

See here: https://github.com/vuejs/vue-cli/blob/dev/CHANGELOG.md#300-beta16-2018-06-08

> env: preserve existing env vars so load in reverse order. (#1503) (7c1ef24)

**Comment by RohanBalraj** [comment]
Same problem running "npm run serve" on Windows 10 using 3.0.0-rc.3. "npm run build" works fine.
Solved this by running "set NODE_ENV=" before running "npm run serve"
